libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -364,45 +313,14 @@ size_t UARTDriver::write(const uint8_t *buffer, size_t size)
         return ret;
     }
 
-    uint16_t _head, space;
-    space = BUF_SPACE(_writebuf);
-    if (space == 0) {
-        return 0;
-    }
-    if (size > space) {
-        size = space;
-    }
-    if (_writebuf_tail < _head) {
-        // perform as single memcpy
-        assert(_writebuf_tail+size <= _writebuf_size);
-        memcpy(&_writebuf[_writebuf_tail], buffer, size);
-        BUF_ADVANCETAIL(_writebuf, size);
-        return size;
-    }
-
-    // perform as two memcpy calls
-    uint16_t n = _writebuf_size - _writebuf_tail;
-    if (n > size) n = size;
-    assert(_writebuf_tail+n <= _writebuf_size);
-    memcpy(&_writebuf[_writebuf_tail], buffer, n);
-    BUF_ADVANCETAIL(_writebuf, n);
-    buffer += n;
-    n = size - n;
-    if (n > 0) {
-        assert(_writebuf_tail+n <= _writebuf_size);
-        memcpy(&_writebuf[_writebuf_tail], buffer, n);
-        BUF_ADVANCETAIL(_writebuf, n);
-    }
-    return size;
+    return _writebuf.write(buffer, size);
```

Comment:
*ByteBuffer::write* returns an *uint32_t* but this function returns a *size_t*. It won't be a problem now because we also write *size_t*, but this should be improved in the future.